sys/ztimer/auto_init.c

```diff
@@ -52,78 +53,201 @@
-ztimer_clock_t *const ZTIMER_MSEC_BASE = &_ztimer_periph_timer_rtt_msec;
-#  define ZTIMER_RTT_INIT (ZTIMER_MSEC_BASE)
-#    if RTT_FREQUENCY != FREQ_1KHZ
-static ztimer_convert_frac_t _ztimer_convert_frac_msec;
-ztimer_clock_t *const ZTIMER_MSEC = &_ztimer_convert_frac_msec.super.super;
-#  define ZTIMER_MSEC_CONVERT_LOWER_FREQ    RTT_FREQUENCY
-#  define ZTIMER_MSEC_CONVERT_LOWER         (&_ztimer_periph_timer_rtt_msec)
-#    else
-ztimer_clock_t *const ZTIMER_MSEC = &_ztimer_periph_timer_rtt_msec;
-#    endif
-#  elif MODULE_ZTIMER_USEC
+#  ifdef ZTIMER_RTT
+ztimer_clock_t *const ZTIMER_MSEC_BASE = &ZTIMER_RTT_CLK;
+#  elif defined(ZTIMER_TIMER)
+ztimer_clock_t *const ZTIMER_MSEC_BASE = &ZTIMER_TIMER_CLK;
+#  else
+#    error No suitable ZTIMER_MSEC config. No rtt or basic timer?
+#  endif
+#  ifdef ZTIMER_MSEC_CONVERT_LOWER_FREQ
 static ztimer_convert_frac_t _ztimer_convert_frac_msec;
 ztimer_clock_t *const ZTIMER_MSEC = &_ztimer_convert_frac_msec.super.super;
-ztimer_clock_t *const ZTIMER_MSEC_BASE = &_ztimer_periph_timer_usec.super;
-#    define ZTIMER_MSEC_CONVERT_LOWER       ZTIMER_USEC_BASE
-#    define ZTIMER_MSEC_CONVERT_LOWER_FREQ  CONFIG_ZTIMER_USEC_BASE_FREQ
 #  else
-#  error No suitable ZTIMER_MSEC config. Maybe add USEMODULE += ztimer_usec?
+ztimer_clock_t *const ZTIMER_MSEC = &ZTIMER_RTT_CLK;
+#   endif
+#endif
+
+#if MODULE_ZTIMER_SEC
+#  ifdef ZTIMER_RTC
+ztimer_clock_t *const ZTIMER_SEC_BASE = &ZTIMER_RTC_CLK;
+ztimer_clock_t *const ZTIMER_SEC = &ZTIMER_RTC_CLK;
+#  elif defined(ZTIMER_RTT)
+ztimer_clock_t *const ZTIMER_SEC_BASE = &ZTIMER_RTT_CLK;
+#  elif defined(ZTIMER_TIMER)
+ztimer_clock_t *const ZTIMER_SEC_BASE = &ZTIMER_TIMER_CLK;
+#  else
+    #  error No suitable ZTIMER_SEC config. No rtc, rtt or basic timer?
+#  endif
+
+#  ifdef ZTIMER_SEC_CONVERT_LOWER_FREQ
+static ztimer_convert_frac_t _ztimer_convert_frac_sec;
+ztimer_clock_t *const ZTIMER_SEC = &_ztimer_convert_frac_sec.super.super;
 #  endif
 #endif
 
 void ztimer_init(void)
 {
-#if MODULE_ZTIMER_USEC
-#  if CONFIG_ZTIMER_USEC_TYPE_PERIPH_TIMER
+/* Step 4: initialize used ztimer-periphery */
+#if USE_ZTIMER_TIMER
     LOG_DEBUG(
-        "ztimer_init(): ZTIMER_USEC using periph timer %u, freq %lu, width %u\n",
+        "ztimer_init(): ZTIMER_TIMER using periph timer %u, freq %lu, width %u\n",
         CONFIG_ZTIMER_USEC_DEV, CONFIG_ZTIMER_USEC_BASE_FREQ,
         CONFIG_ZTIMER_USEC_WIDTH);
-
-    ztimer_periph_timer_init(&_ztimer_periph_timer_usec, CONFIG_ZTIMER_USEC_DEV,
+    ztimer_periph_timer_init(&ZTIMER_TIMER, CONFIG_ZTIMER_USEC_DEV,
                              CONFIG_ZTIMER_USEC_BASE_FREQ,
                              WIDTH_TO_MAXVAL(CONFIG_ZTIMER_USEC_WIDTH));
+#  ifdef MODULE_PM_LAYERED
+    LOG_DEBUG("ztimer_init(): ZTIMER_TIMER setting required_pm_mode to %i\n",
+              CONFIG_ZTIMER_TIMER_REQUIRED_PM_MODE);
+    ZTIMER_TIMER_CLK.required_pm_mode = CONFIG_ZTIMER_TIMER_REQUIRED_PM_MODE;
 #  endif
+#endif
+
+#ifdef USE_ZTIMER_RTT
```

Comment:
```suggestion
#if USE_ZTIMER_RTT
```

Maybe? This would allow defining it two zero when needed for quick and dirty debugging.

Reply:
the possibility to define it to 0 for debugging purposes is a good find and of cause the error message erorros. I will change and squash

Reply:
and back again this will not work the way you intend to use it since `USE_ZTIMER_*` is an internal marker whether to alloc (static) space for the coresponing hardware the availability is decided by the use of the module leading to the name (ZTIMER_*) being defined.
but i change it for consitency

Reply:
this made me think about the naming of the `USE_ZTIMER...` defines i think they should be renamed `INIT_ZTiMMER...`

Reply:
DONE